Source/Events.Processing/Filters/FiltersService.cs

```diff
@@ -174,5 +307,107 @@ bool TryStartStreamProcessors(IEnumerable<(TenantId, FilterRegistrationResult<Re
             IReverseCallDispatcher<FiltersClientToRuntimeMessage, FilterRuntimeToClientMessage, FiltersRegistrationRequest, FilterRegistrationResponse, FilterEventRequest, FilterResponse> dispatcher,
             Failure failure,
             CancellationToken cancellationToken) => dispatcher.Reject(new FilterRegistrationResponse { Failure = failure }, cancellationToken);
+
+        bool PartitionedTryStartStreamProcessors(IEnumerable<(TenantId, FilterRegistrationResult<RemoteFilterDefinition>)> registrationResults, out Failure failure)
+        {
+            failure = null;
+            var failedRegistrationReasons = registrationResults
+                                                .Where(tenantAndResult => !tenantAndResult.Item2.Succeeded)
+                                                .Select(tenantAndResult => $"For tenant '{tenantAndResult.Item1}':\n\t{string.Join("\n\t", tenantAndResult.Item2.FailureReason.Value.Split("\n"))}");
+            if (failedRegistrationReasons.Any())
+            {
+                var failureMessage = $"Failed to register filter:\n\t";
+                failureMessage += string.Join("\n\t", failedRegistrationReasons);
+                failure = new Failure(FiltersFailures.FailedToRegisterFilter, failureMessage);
+                return false;
+            }
+
+            foreach ((var tenant, var result) in registrationResults)
+            {
+                _ = result.FilterStreamProcessor.Start();
+            }
+
+            return true;
+        }
+
+        async Task<IEnumerable<(TenantId, FilterRegistrationResult<RemoteFilterDefinition>)>> PartitionedRegisterStreamProcessorsForAllTenants(
+            ScopeId scope,
+            StreamId sourceStream,
+            StreamId targetStream,
+            IReverseCallDispatcher<PartitionedFiltersClientToRuntimeMessage, FilterRuntimeToClientMessage, PartitionedFiltersRegistrationRequest, FilterRegistrationResponse, FilterEventRequest, PartitionedFilterResponse> dispatcher,
+            CancellationToken cancellationToken)
+        {
+            var registrationResults = new List<(TenantId, FilterRegistrationResult<RemoteFilterDefinition>)>();
+            foreach (var tenant in _tenants.All)
+            {
+                _executionContextManager.CurrentFor(tenant);
+                var filterProcessor = new Partitioned.FilterProcessor(
+                    scope,
+                    new RemoteFilterDefinition(sourceStream, targetStream),
+                    dispatcher,
+                    _getEventsToStreamsWriter(),
+                    _logger);
+                var filters = _getFilters();
+                var registrationResult = await filters.Register(filterProcessor, cancellationToken).ConfigureAwait(false);
+                registrationResults.Add((tenant, registrationResult));
+            }
+
+            return registrationResults.AsEnumerable();
```

Comment:
The `List<..>` implements `IEnumerable` so no need to convert it.
```suggestion
            return registrationResults;
```